modules/partition-replicator/src/main/java/org/apache/ignite/internal/partition/replicator/raft/snapshot/metrics/RaftSnapshotsMetricsSource.java

```diff
@@ -0,0 +1,297 @@
+
+    private final AtomicInteger snapshotsPreparingIndexForBuildCounter = new AtomicInteger();
+
+    private final AtomicInteger snapshotsLoadingMvDataCounter = new AtomicInteger();
+
+    private final AtomicInteger snapshotsLoadingTxMetaCounter = new AtomicInteger();
+
+    private final AtomicInteger totalOutgoingSnapshotsCounter = new AtomicInteger();
+
+    /**
+     * Creates a new metric source with the name {@code raft.snapshots}.
+     */
+    public RaftSnapshotsMetricsSource() {
+        super("raft.snapshots");
+    }
+
+    @Override
+    protected Holder createHolder() {
+        return new Holder(
+                totalIncomingSnapshotsCounter::get,
+                snapshotsLoadingMetaCounter::get,
+                snapshotsWaitingCatalogCounter::get,
+                snapshotsPreparingStoragesCounter::get,
+                snapshotsPreparingIndexForBuildCounter::get,
+                snapshotsLoadingMvDataCounter::get,
+                snapshotsLoadingTxMetaCounter::get,
+                totalOutgoingSnapshotsCounter::get
+        );
+    }
+
+    /**
+     * Marks the start of an incoming snapshot installation.
+     * Increments the {@code TotalIncomingSnapshots} counter.
+     */
+    public void onSnapshotInstallationStart() {
+        totalIncomingSnapshotsCounter.incrementAndGet();
+    }
+
+    /**
+     * Marks the end of an incoming snapshot installation.
+     * Decrements the {@code TotalIncomingSnapshots} counter.
+     */
+    public void onSnapshotInstallationEnd() {
+        totalIncomingSnapshotsCounter.decrementAndGet();
+    }
+
+    /**
+     * Marks the beginning of the "load snapshot metadata" phase during incoming snapshot installation.
+     * Increments the {@code IncomingSnapshotsLoadingMeta} counter.
+     */
+    public void onLoadSnapshotPhaseStart() {
+        snapshotsLoadingMetaCounter.incrementAndGet();
+    }
+
+    /**
+     * Marks the end of the "load snapshot metadata" phase during incoming snapshot installation.
+     * Decrements the {@code IncomingSnapshotsLoadingMeta} counter.
+     */
+    public void onLoadSnapshotPhaseEnd() {
+        snapshotsLoadingMetaCounter.decrementAndGet();
+    }
+
+    /**
+     * Marks the beginning of the phase where the node waits for catalog to be ready/apply updates
+     * for the incoming snapshot installation.
+     * Increments the {@code IncomingSnapshotsWaitingCatalog} counter.
+     */
+    public void onWaitingCatalogPhaseStart() {
+        snapshotsWaitingCatalogCounter.incrementAndGet();
+    }
+
+    /**
+     * Marks the end of the "waiting for catalog" phase during incoming snapshot installation.
+     * Decrements the {@code IncomingSnapshotsWaitingCatalog} counter.
+     */
+    public void onWaitingCatalogPhaseEnd() {
+        snapshotsWaitingCatalogCounter.decrementAndGet();
+    }
+
+    /**
+     * Marks the beginning of the phase where MV (multi-version) data is loaded from the snapshot.
```

Comment:
'from the snapshot' sounds weird. It is a part of the snapshot. Maybe 'where snapshot MV data is loaded'?